pkg/sbproxy/notifications/producer.go

```diff
@@ -302,13 +300,24 @@ func (p *Producer) connect(ctx context.Context) error {
 		if resp != nil {
 			log.C(ctx).WithError(err).Errorf("Could not connect to %s: status: %d", &connectURL, resp.StatusCode)
 			if resp.StatusCode == http.StatusGone {
+				p.lastNotificationRevision = unknownRevision
 				return errLastNotificationGone
 			}
 		}
+
 		return err
 	}
+
+	// The websocket connection was successfully established but with unknown revision, therefore trigger resync
+	if p.lastNotificationRevision == unknownRevision {
+		log.C(ctx).Info("Websocket connection established with unknown revision. Triggering resync...")
+		resyncChan <- struct{}{}
```

Comment:
connect does multiple things now. Dunno if the readability is so bad now, that we'd rather violate solid